src/components/table/table.stories.js

```diff
@@ -112,8 +103,15 @@ const buildRows = ({ pageSize, totalRecords }) => {
   );
 };
 
+const roundPageValue = (value) => {
+  const roundedValue = Math.round(value);
+  if (!value || roundedValue < 1) return '1';
+  return roundedValue;
+};
+
 const setCurrentPage = ({ totalRecords, pageSize, paginate }) => {
-  const maxValidPage = Math.round(totalRecords / pageSize) + (totalRecords % pageSize);
+  const roundedValue = roundPageValue(totalRecords / pageSize);
+  const maxValidPage = roundedValue < totalRecords / pageSize ? roundedValue + 1 : roundedValue;
```

Comment:
Might not an idea to put some brackets into this line to make it a bit clearer what is happening.

Reply:
Total pages could be expressed as follows:

```
Math.max(Math.ceil(totalRecords / pageSize), 1)
```
you would want to check that `pageSize` is set and if it's not then return `1`.

Reply:
or create a const containing `totalRecords / pageSize`

Reply:
I went with a mix of both!